### ISSUE TITLE
Get conscrypt working with Netty

### ISSUE THREAD

**Issue by nmittler** [grpc/grpc-java#1927]
https://android.googlesource.com/platform/external/conscrypt/

& issues

https://code.google.com/p/android/issues/list?can=2&q=conscrypt&colspec=ID+Status+Priority+Owner+Summary+Stars+Reporter+Opened&cells=tiles

FYI @ejona86 @normanmaurer @louiscryan @Scottmitch @buchgr 

Related to: https://github.com/netty/netty-tcnative/issues/114

**Comment by nmittler** [comment]
FYI - I have a first cut at getting conscrypt to build for non-Android: https://android-review.googlesource.com/#/c/289828/

**Comment by nmittler** [comment]
@ejona86 @normanmaurer @louiscryan @Scottmitch @buchgr

After some hacking, I've got the basic Netty ALPN tests working with conscrypt in [this branch](https://github.com/nmittler/netty/tree/conscrypt) ([commit](https://github.com/nmittler/netty/commit/01c3653dc9d8932df2a7a568ea34d84ef39055f5)).

This is all preliminary, since the required changes don't yet exist in the Conscrypt repository ... but it looks promising!

**Comment by nmittler** [comment]
Just created [a grpc branch](https://github.com/nmittler/grpc-java/tree/conscrypt) ([commit](https://github.com/nmittler/grpc-java/commit/ab752a6e7451c452b4c0dca10f5ca9e9133a2627)) to benchmark conscrypt.

The results look pretty shaky for conscrypt. I still need to investigate why it's performing so poorly:

```
JRE: Sun 1.8.0_92
Cipher used: One of "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
                    "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
                    "TLS_DHE_RSA_WITH_AES_128_GCM_SHA256" 

JDK:
Channels:                       4
Outstanding RPCs per Channel:   10
Server Payload Size:            0
Client Payload Size:            0
50%ile Latency (in micros):     947
90%ile Latency (in micros):     1439
95%ile Latency (in micros):     1775
99%ile Latency (in micros):     7295
99.9%ile Latency (in micros):   16383
Maximum Latency (in micros):    38655
QPS:                            34566

netty-tcnative:
Channels:                       4
Outstanding RPCs per Channel:   10
Server Payload Size:            0
Client Payload Size:            0
50%ile Latency (in micros):     607
90%ile Latency (in micros):     947
95%ile Latency (in micros):     1207
99%ile Latency (in micros):     5983
99.9%ile Latency (in micros):   15743
Maximum Latency (in micros):    38655
QPS:                            50721

Conscrypt:
Channels:                       4
Outstanding RPCs per Channel:   10
Server Payload Size:            0
Client Payload Size:            0
50%ile Latency (in micros):     1239
90%ile Latency (in micros):     1911
95%ile Latency (in micros):     2351
99%ile Latency (in micros):     8383
99.9%ile Latency (in micros):   17919
Maximum Latency (in micros):    54783
QPS:                            26950
```

**Comment by ejona86** [comment]
Fixed by #3401
